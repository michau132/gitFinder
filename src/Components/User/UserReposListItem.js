import React from 'react';
import PropTypes from 'prop-types';
import {
  ListItem,
  Checkbox,
  ListItemText,
  ListItemSecondaryAction,
  Button,
  Grid,
  Typography,
  Collapse,
} from '@material-ui/core';
import styled from 'styled-components';

const StyledCollapse = styled(Collapse)`
  ${({ isfounded }) => (isfounded ? 'background-color: rgba(0, 0, 0, 0.24)' : '')}
`;

const UserReposListItem = ({
  listItem: {
    id,
    name,
    forks_count,
    stargazers_count,
    description,
    html_url,
    isChecked = false,
    isFounded = false,
    isHidden = false,
  },
  handleSelectUserRepo,
  handleHideSingleRepo,
}) => (
  <StyledCollapse
    in={!isHidden}
    timeout={700}
    isfounded={isFounded ? 1 : 0}
  >
    <ListItem
      divider
      button
      onClick={() => handleSelectUserRepo(id)}
      selected={isChecked}
      className="listItem"
    >
      <Checkbox
        checked={isChecked}
        color="primary"
        readOnly
        className="checkbox"
      />
      <Grid container direction="column">
        <ListItemText>
          <Typography variant="subtitle2" gutterBottom className="forkAndCount">
            {`${name}(forks: ${forks_count}, stars${stargazers_count})`}
          </Typography>
        </ListItemText>
        <ListItemText primary={description} className="description" />
      </Grid>
      <ListItemSecondaryAction>
        <Button
          variant="contained"
          color="primary"
          target="_blank"
          href={html_url}
          className="open"
        >
        open
        </Button>
        <Button
          variant="contained"
          color="primary"
          onClick={() => handleHideSingleRepo(id)}
          className="hide"
        >
        hide
        </Button>
      </ListItemSecondaryAction>
    </ListItem>
  </StyledCollapse>
);

UserReposListItem.defaultProps = {
  listItem: {
    name: null,
    html_url: null,
  },
};

UserReposListItem.propTypes = {
  listItem: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    forks_count: PropTypes.number,
    stargazers_count: PropTypes.number,
    description: PropTypes.string,
    html_url: PropTypes.string.isRequired,
    isChecked: PropTypes.bool,
    isFounded: PropTypes.bool,
    isHidden: PropTypes.bool,
  }),
  handleSelectUserRepo: PropTypes.func.isRequired,
  handleHideSingleRepo: PropTypes.func.isRequired,
};

export default UserReposListItem;
