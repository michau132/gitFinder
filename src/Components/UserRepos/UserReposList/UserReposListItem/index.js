/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './style.css';
import Button from '../../../Button';

const UserReposListItem = ({
  listItem,
  selectUserRepo,
  hideSingleRepo,
  openSingleRepo,
}) => (
  <li className={classnames(styles.listItem, { [styles.active]: listItem.isChecked })}>
    <div className={styles.listItemInfo} onClick={() => selectUserRepo(listItem.id)}>
      <div className={styles.infoWrapper}>
        <input type="checkbox" checked={listItem.isChecked || false} readOnly />
        <h4 className={styles.infoName}>
          {listItem.name}
          (forks:
          {' '}
          {listItem.forks_count}
          ,
          stars:
          {' '}
          {listItem.stargazers_count}
          )
        </h4>
      </div>
      <p className={styles.infoDesc}>{listItem.description}</p>
    </div>
    <div className={styles.btnWrapper}>
      <Button
        type="button"
        onClick={() => openSingleRepo(listItem.html_url)}
        text="open"
      />
      <Button
        type="button"
        onClick={() => hideSingleRepo(listItem.id)}
        text="hide"
      />
    </div>
  </li>
);

UserReposListItem.defaultProps = {
  listItem: {
    name: null,
    id: null,
    html_url: null,
  },
};

UserReposListItem.propTypes = {
  listItem: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    forks_count: PropTypes.number,
    stargazers_count: PropTypes.number,
    description: PropTypes.string,
    html_url: PropTypes.string.isRequired,
  }),
  selectUserRepo: PropTypes.func.isRequired,
  hideSingleRepo: PropTypes.func.isRequired,
  openSingleRepo: PropTypes.func.isRequired,
};

export default UserReposListItem;
