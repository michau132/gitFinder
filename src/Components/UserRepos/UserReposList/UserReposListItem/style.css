.listItem {
  margin: 10px 0;
  display: flex;
  justify-content: space-between;
}

.listItem:hover {
  background-color: #c0c79f;
}


.listItemInfo {
  flex-grow: 1;
  cursor: pointer;
  max-width: 80%;
}

.infoName {
  display: inline-block;
  margin: 10px;
}

.infoWrapper {
  overflow: hidden;
  flex: 1;
}

.infoDesc {
  margin: 0;
  word-wrap: break-word;
}

.btnWrapper {
  margin-right: 15px;
  align-self: center;
}

.btnOpen {
  text-decoration: none;
  text-transform: uppercase;
  margin-right: 10px;
  padding: 5px 10px;
  background: #fff;
  border: 2px solid #7619FF;
  color: #7619FF;
  transition: 0.2s all;
}

.btnOpen:hover {
  background: #43d1af;
  color: #fff;
}

.btnHide {
  text-transform: uppercase;
  padding: 5px 10px;
  background: #fff;
  border: 2px solid #7619FF;
  color: #7619FF;
  transition: 0.2s all;
}

.btnHide:hover {
  background: #c22121;
  color: #fff;
}

.isFounded {
  background-color: violet;
}

.isChecked {
  background-color: #FFB10D;
}

.isChecked.isFounded {
  background-color: lightskyblue;
}

.isChecked.isFounded:hover {
  background-color: silver;
}

.isHidden {
  display: none;
}
