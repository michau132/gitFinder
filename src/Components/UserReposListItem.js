import React from 'react';
import PropTypes from 'prop-types';

const UserReposListItem = ({ listItem }) => (
  <li style={{ paddingTop: '10px' }}>
    <b>
      <i>
        Name repo:
        {listItem.name}
      </i>
    </b>
    <br />
    <b>modified:</b>
    {' '}
    {listItem.pushed_at}
  </li>
);

UserReposListItem.defaultProps = {
  listItem: {
    name: null,
    pushed_at: null,
  },
};

UserReposListItem.propTypes = {
  listItem: PropTypes.shape({
    name: PropTypes.string.isRequired,
    pushed_at: PropTypes.string.isRequired,
  }),
};

export default UserReposListItem;
