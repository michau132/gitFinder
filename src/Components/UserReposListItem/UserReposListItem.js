import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';

const UserReposListItem = ({ listItem }) => (
  <li className={styles.listItem}>
    <div className={styles.listItemInfo}>
      <div className={styles.infoWrapper}>
        <input type="checkbox" />
        <h4 className={styles.infoName}>
          {listItem.name}
          (forks:
          {' '}
          {listItem.forks_count}
          ,
          stars:
          {' '}
          {listItem.stargazers_count}
          )
        </h4>
      </div>
      <p className={styles.infoDesc}>{listItem.description}</p>
    </div>
    <div className={styles.btnWrapper}>
      <a href={listItem.html_url} target="blank">open</a>
      <button type="button">hide</button>
    </div>
  </li>
);

UserReposListItem.defaultProps = {
  listItem: {
    name: null,
    pushed_at: null,
  },
};

UserReposListItem.propTypes = {
  listItem: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
};

export default UserReposListItem;
