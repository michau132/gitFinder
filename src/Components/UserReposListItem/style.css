.listItem {
  margin: 10px 0;
  display: flex;
  justify-content: space-between;
}

.infoName {
  display: inline-block;
  margin: 10px;
}
.infoWrapper {
  overflow: hidden;
  flex: 1;
}
.infoDesc {
  margin: 0;
  word-wrap: break-word;
}
.btnWrapper {
  margin-right: 15px;
}