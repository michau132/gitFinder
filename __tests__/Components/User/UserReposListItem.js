import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserReposListItem from '../../../src/Components/User/UserReposListItem';

describe('testing UserReposListItem component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      listItem: {
        id: 1,
        name: 'firstRepo',
        forks_count: 0,
        stargazers_count: 0,
        description: 'First repo description',
        html_url: 'https://github.com/aOko123/firstRepo',
      },
      handleSelectUserRepo: jest.fn(),
      handleHideSingleRepo: jest.fn(),
    };

    wrapper = shallow(<UserReposListItem {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('wrapper should be visible', () => {
    expect(wrapper.prop('in')).toBe(true);
  });

  test('wrapper should not have background color, it is not founded', () => {
    expect(wrapper.prop('isfounded')).toBeFalsy();
  });

  test('wrapper should not have been selected', () => {
    const listItem = wrapper.find('.listItem');
    expect(listItem.prop('selected')).toBeFalsy();
  });

  test('checkbox should not have been selected', () => {
    const checkbox = wrapper.find('.checkbox');
    expect(checkbox.prop('checked')).toBeFalsy();
  });

  test('calls event handleSelectUserRepo', () => {
    const listItem = wrapper.find('.listItem');
    listItem.simulate('click');
    expect(props.handleSelectUserRepo).toHaveBeenCalledWith(props.listItem.id);
  });

  test('Button should be defined and redirect to user repository', () => {
    const button = wrapper.find('.open');
    expect(button.prop('href')).toBe(props.listItem.html_url);
  });

  test('calls event handleHideSingleRepo', () => {
    const button = wrapper.find('.hide');
    button.simulate('click');
    expect(props.handleHideSingleRepo).toHaveBeenCalledWith(props.listItem.id);
  });
});
