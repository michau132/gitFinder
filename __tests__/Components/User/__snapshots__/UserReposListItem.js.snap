// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserReposListItem component matching snapshot 1`] = `
<Styled(WithStyles(Collapse))
  in={true}
  isfounded={0}
  timeout={700}
>
  <WithStyles(ListItem)
    button={true}
    className="listItem"
    divider={true}
    onClick={[Function]}
    selected={false}
  >
    <WithStyles(Checkbox)
      checked={false}
      className="checkbox"
      color="primary"
      readOnly={true}
    />
    <WithStyles(Grid)
      container={true}
      direction="column"
    >
      <WithStyles(ListItemText)>
        <WithStyles(Typography)
          className="forkAndCount"
          gutterBottom={true}
          variant="subtitle2"
        >
          firstRepo(forks: 0, stars0)
        </WithStyles(Typography)>
      </WithStyles(ListItemText)>
      <WithStyles(ListItemText)
        className="description"
        primary="First repo description"
      />
    </WithStyles(Grid)>
    <WithStyles(ListItemSecondaryAction)>
      <WithStyles(Button)
        className="open"
        color="primary"
        href="https://github.com/aOko123/firstRepo"
        target="_blank"
        variant="contained"
      >
        open
      </WithStyles(Button)>
      <WithStyles(Button)
        className="hide"
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        hide
      </WithStyles(Button)>
    </WithStyles(ListItemSecondaryAction)>
  </WithStyles(ListItem)>
</Styled(WithStyles(Collapse))>
`;
